react/pages/feed/EmptyConnectTwitter/index.js

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from 'react';
+import styled from 'styled-components';
+import { graphql, compose } from 'react-apollo';
+import axios from 'axios';
+import { propType } from 'graphql-anywhere';
+import PropTypes from 'prop-types';
+import sharify from 'sharify';
+
+import Modal from 'react/components/UI/Modal';
+import Text from 'react/components/UI/Text';
+import PageContainer from 'react/components/UI/PageContainer';
+import ConnectTwitter from 'react/components/ConnectTwitter/index';
+import { GenericButton } from 'react/components/UI/GenericButton';
+import updateFlagMutation from 'react/pages/feed/EmptyConnectTwitter/mutations/index';
+
+import EmptyFeedConnectTwitterQuery from 'react/pages/feed/EmptyConnectTwitter/queries/index';
+import EmptyFeedCheckFragment from 'react/pages/feed/EmptyConnectTwitter/fragments/index';
+
+const ActionContainer = styled.div`
+  text-align: center;
+  padding: ${x => x.theme.space[7]} 0;
+`;
+
+const Headline = styled(Text).attrs({
+  fontSize: 7,
+  lineHeight: 1,
+  align: 'center',
+})`
+`;
+
+const SmallText = styled(Text).attrs({
+  fontSize: 3,
+  lineHeight: 1,
+  mt: 4,
+})``;
+
+const Link = styled(SmallText)`
+  text-decoration: underline;
+  cursor: pointer;
+  display: inline-block;
+`;
+
+class EmptyConnectTwitterPage extends Component {
+  static propTypes = {
+    data: PropTypes.shape({
+      me: propType(EmptyFeedCheckFragment),
+    }).isRequired,
+    showModal: PropTypes.bool,
+    updateFlag: PropTypes.func.isRequired,
+  }
+
+  static defaultProps = {
+    showModal: false,
+  }
+
+  updateFlagAndRefresh = (e) => {
+    e.preventDefault();
+
+    const { updateFlag } = this.props;
+
+    updateFlag({
+      variables: {
+        name: 'HAS_SEEN_FEED_CONNECT_TWITTER',
+        value: true,
+      },
+    })
+      .then(() => { axios.get('/me/refresh'); })
+      .then(() => { setTimeout(() => { window.location = '/feed'; }, 400); });
```

Comment:
Why the setTimeout?

Reply:
Sigh, to let the cache clear. I'm actually still having issues where occasionally after "Done" is pressed the page refreshes and this message is still there, so there has to be a better way.

Reply:
Which cache?

Reply:
Passport / user cache.

Reply:
So can we just fix that route then?

Reply:
I really don't want setTimeouts littered around; there shouldn't be anything preventing us from OKing on the callback to clear whatever it's doing.